concordia/tests/test_views.py

```diff
@@ -1088,6 +1088,59 @@ def test_duplicate_tag_submission(self):
         self.assertEqual(["bar", "foo", "quux"], data["user_tags"])
         self.assertEqual(["baaz", "bar", "foo", "quux"], data["all_tags"])
 
+    def test_tag_deletion(self):
+        asset = create_asset()
+        self.login_user()
+
+        test_tags = ["foo", "bar"]
+        resp = self.client.post(
+            reverse("submit-tags", kwargs={"asset_pk": asset.pk}),
+            data={"tags": test_tags},
+        )
+        test_tags = [
+            "foo",
+        ]
+        resp = self.client.post(
+            reverse("submit-tags", kwargs={"asset_pk": asset.pk}),
+            data={"tags": test_tags},
+        )
+        data = self.assertValidJSON(resp, expected_status=200)
+        self.assertIn("user_tags", data)
+        self.assertIn("all_tags", data)
+
+        self.assertEqual(sorted(test_tags), data["user_tags"])
+        self.assertEqual(sorted(test_tags), data["all_tags"])
+
+    def test_tag_deletion_with_multiple_users(self):
+        asset = create_asset()
+        self.login_user()
+        test_tags = ["foo", "bar"]
+        resp = self.client.post(
+            reverse("submit-tags", kwargs={"asset_pk": asset.pk}),
+            data={"tags": test_tags},
+        )
+        data = self.assertValidJSON(resp, expected_status=200)
+        self.assertIn("user_tags", data)
+        self.assertIn("all_tags", data)
+        self.assertEqual(sorted(test_tags), data["user_tags"])
+        self.assertEqual(sorted(test_tags), data["all_tags"])
+
+        self.client.logout()
+
+        self.login_user("tester")
```

Comment:
Isn't this the default name, meaning that both of these tests are running as `tester`? I would make this clear in the test suite by using different usernames explicitly in both calls — `self.login_user("first_user")`, `self.login_user("second_user")`, etc. — so there's no chance of a test appearing to work by accident. That could be enforced by doing a check like ensuring that `asset.userassettagcollection_set.user_set.count()` is 2.